nan.h

```diff
@@ -29,6 +29,9 @@
 #include <climits>
 #if defined(_MSC_VER)
 # pragma warning( disable : 4530 )
+# if _MSC_VER < 1700  // < VS 2012, don't know when the warning stops appearing
+#  pragma warning( disable : 4344 )
```

Comment:
I agree that C4344 is an annoying warning but I don't think disabling it file-wide is a good thing.